phalcon/Mvc/Model/MetaData.zep

```diff
@@ -525,21 +525,13 @@ abstract class MetaData implements InjectionAwareInterface, MetaDataInterface
      */
     final public function readColumnMap(<ModelInterface> model) -> array | null
     {
-        var keyName, data;
+        var keyName;
 
         if !globals_get("orm.column_renaming") {
             return null;
         }
-
-        let keyName = get_class_lower(model);
-
-        if !fetch data, this->columnMap[keyName] {
-            this->initialize(model, null, null, null);
-
-            let data = this->columnMap[keyName];
-        }
-
-        return data;
+        let keyName = this->getColumnMapUniqueKey(model);
+        return this->columnMap[keyName];
```

Comment:
This might generate notice error. Use ternary condition:
```
isset(this->columnMap[keyName]) ? this->columnMap[keyName] : null;
```

Reply:
actually we only need to check if the keyName is not null.

Reply:
What if the key does not exist? That would throw a warning no?

Reply:
it already throws errors in other places 
The model does this
```
        let dataTypes           = metaData->getDataTypes(this),
            bindDataTypes       = metaData->getBindTypes(this),
            nonPrimary          = metaData->getNonPrimaryKeyAttributes(this),
            automaticAttributes = metaData->getAutomaticUpdateAttributes(this);
```

and all of them throws exceptions in the metadata.
```
        if unlikely typeof data != "array" {
            throw new Exception("The meta-data is invalid or is corrupt");
        }
```

do you want me to send warnings to php?

Reply:
Throwing exceptions is fine (for instance if `data` is not an array as shown above). I would prefer an extra check to happen if the key of the array does not exist.